urla05239b71c_viniciusinnovatium_mumps2java_api_tgz-pJ8-com_disclinc_netmanager_script_test_ExecTesteNewJ8/src/mLibrary/mCmd.java

```diff
@@ -0,0 +1,261 @@
+	/*
+	 * Pausa o processamento por um determinado número de milisegundos
+	 */
+	public void Hang(Object obj) {
+		try {
+			Thread.sleep(Long.valueOf(String.valueOf(obj)));
+		} catch (NumberFormatException e) {
+			Logger.getLogger(mClass.class.getName()).log(Level.SEVERE, null, e);
+		} catch (InterruptedException e) {
+			Logger.getLogger(mClass.class.getName()).log(Level.SEVERE, null, e);
+		}
+
+	}
+
+	public boolean isIndirectionExecution(String content) {
+		if (content == null) {
+			return false;
+		}
+		return content.indexOf("(") != -1;
+	}
+
+	public boolean isMethodExecution(String content) {
+		if (content == null) {
+			return false;
+		}
+		return content.indexOf("^") != -1;
+	}
+
+	public void Job(String string, String string2) {
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException();
+	}
+
+	public void Lock(String string, String string2, String string3) {
+		// TODO REVISAR IMPLEMENTAÇÃO
+	}
+
+	public void LockInc(mVar var, int i) {
+		// TODO REVISAR IMPLEMENTAÇÃO
+	}
+
+	public void Merge(mVar target, mVar source) {
+		m$.merge(target, source);
+		// TODO Auto-generated method stub
+	}
+
+	public void Open(Object $$$OprLog) {
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException();
+	}
+
+	public void Open(Object object, String string, int i) {
+		// TODO Auto-generated method stub
+
+	}
+
+	public void Open(Object object, String string, Object concat, int i) {
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException();
+	}
+
+	/*
+	 * Exibe mensagem no console solicitando entrada de dados.
+	 */
+	public void Read(Object... parameters) {
+		Scanner s = null;
+		try {
+			String variable = "";
+			if (parameters != null && parameters.length > 0) {
+				variable = String.valueOf(parameters[0]);
+			}
+			s = new Scanner(System.in);
+			System.out.println(variable);
+			String input = s.next();
+			System.out.println("Input: " + input);
+		} catch (Exception e) {
+			Logger.getLogger(mClass.class.getName()).log(Level.SEVERE, null, e);
+		} finally {
+			if (s != null) {
+				s.close();
+			}
+		}
+	}
+
+	public void SQL() {
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException();
+	}
+
+	public void Unlock(mVar var) {
+		// TODO REVISAR IMPLEMENTAÇÃO
+	}
+
+	public void Unlock(mVar var, String str) {
+		// TODO REVISAR IMPLEMENTAÇÃO
+	}
+
+	public void Unlock(String string) {
+		// TODO REVISAR IMPLEMENTAÇÃO
+	}
+
+	public void Use(Object $$$OprLog) {
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException();
+	}
+
+	/*
+	 * Remove toString to wirte faster
+	 */
+	@TODO
+	public void Write(Object... string) {
+		try {
+			Writer writer = m$.getWriter();
+			for (Object str : string) {
+				try {
+					writer.write(String.valueOf(str));
+					if(String.valueOf(str).contains("<HR")){
+						System.out.print("");
+					}
+					System.out.print(mFncUtil.toString(str));
+				} catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)